MoreLinq.Test/MoveRangeTest.cs

```diff
@@ -0,0 +1,121 @@
+﻿#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2017 Leandro F. Vieira (leandromoh). All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using System.Linq;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class MoveRangeTest
+    {
+        [Test]
+        public void MoveRangeWithNegativeIndex()
+        {
+            Assert.ThrowsArgumentOutOfRangeException("oldIndex", () =>
+                new[] { 1 }.MoveRange(-1, 0, 0));
+        }
+
+        [Test]
+        public void MoveRangeWithNegativeCount()
+        {
+            Assert.ThrowsArgumentOutOfRangeException("count", () =>
+                new[] { 1 }.MoveRange(0, -1, 0));
+        }
+
+        [Test]
+        public void MoveRangeWithNegativeNewIndex()
+        {
+            Assert.ThrowsArgumentOutOfRangeException("newIndex", () =>
+                new[] { 1 }.MoveRange(0, 0, -1));
+        }
+
+        [Test]
+        public void MoveRangeIsLazy()
+        {
+            new BreakingSequence<int>().MoveRange(0, 0, 0);
+        }
+
+        [TestCase(10, 5,  3)]
+        [TestCase(10, 3,  5)]
+        [TestCase(10, 0,  5)]
+        [TestCase(10, 5,  0)]
+        [TestCase(10, 6,  1)]
+        [TestCase(10, 1,  6)]
+        [TestCase(10, 0, 10)]
+        [TestCase(10, 10, 0)]
+        [TestCase(10, 3, 10)]
+        [TestCase(10, 10, 3)]
+        [TestCase(10, 99, 2)]
+        public void MoveRange(int length, int index, int count)
```

Comment:
It's not obvious what's going on here really. The `length` argument is always 10 and there are no expectations in the test data. Later in the test method, expectations are generated dynamically so there's mix of strategies here. If you want to generate test cases, I recommend you do so with a [test case source](https://github.com/nunit/docs/wiki/TestCaseData), or see the [`ConcatMany` test](https://github.com/atifaziz/MoreLINQ/blob/1afdf8394fa18ce6af146b7a816d6843f62616b4/MoreLinq.Test/ConcatTest.cs#L101-L119) for an example.